receiver/kubeletstatsreceiver/scraper.go

```diff
@@ -57,6 +57,31 @@ func newKubeletScraper(
 	metricsConfig metadata.MetricsBuilderConfig,
 	nodeName string,
 ) (scraper.Metrics, error) {
+	if EnableCPUUsageMetrics.IsEnabled() {
```

Comment:
this logic was moved to scraper due to easier testability -> all tests are using `newKubeletScraper()` function to initialize the scraper and test the exported metrics and therefore if the logic is present in the factory, there is no possibility to test the new behavior.

Reply:
How about moving this check to the config validation?

Reply:
We dont want to do that bc the result of the gate may change the config and generally it is a bad practice to modify the config in the validate function

```
                 metricsConfig.Metrics.ContainerCPUUtilization.Enabled = true
		metricsConfig.Metrics.K8sPodCPUUtilization.Enabled = true
		metricsConfig.Metrics.K8sNodeCPUUtilization.Enabled = true
```